src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -198,6 +198,15 @@ bool concat_in_place_optimization::match(concatenation_node& node) {
         if (input.first->get_users().size() > 2)
             return false;
 
+        // Check siblings whether they are oneDNN impl.
```

Comment:
`// If sibling is using onednn impl and batch > 1, the onednn impl cannot process the implicit concat'ed buffer. Onednn impls can process implicit concat'ed buffer only through buffer pointer manipulation`

Reply:
Added